### PR TITLE
samples: nrf9160: modem_shell: port to nrf_modem_at APIs

### PR DESCRIPTION
**Pull request by lemrey**
Port to new AT APIs

### CODE REVIEW

**Review by hiltunent — APPROVED** [review]

File: samples/nrf9160/modem_shell/src/main.c [hunk 1]
```diff
@@ -143,7 +142,6 @@ void main(void)
 	}
 	printk("Initialized modemlib\n");
 
-	at_cmd_init();
```
Comment (trantanen) [inline samples/nrf9160/modem_shell/src/main.c:146]: I guess we cannot remove this before all dependencies have been converted to use `nrf_modem_at.h` and `at_monitor.h`?
At least SMS and modem_info libraries haven't been changed and SMS doesn't work with this modem_shell branch.

PR for SMS: https://github.com/nrfconnect/sdk-nrf/pull/6314

**Review by trantanen — CHANGES_REQUESTED** [review]

File: samples/nrf9160/modem_shell/src/main.c (hunk 1 above)
Comment (lemrey) [reply samples/nrf9160/modem_shell/src/main.c:146]: Yep, there's the SMS and `modem_info` which I am porting right now. Are there more dependencies?

**Review by tokangas — APPROVED** [review]

File: samples/nrf9160/modem_shell/src/main.c (hunk 1 above)
Comment (trantanen) [reply samples/nrf9160/modem_shell/src/main.c:146]: Based on what I could find from nrf, `lwm2m` is likely the only additional one.
I guess lwm2m has dependency to SMS, which will take some time due to the linking bug in Zephyr (which prevents Unity tests from passing):
https://github.com/nrfconnect/sdk-nrf/pull/6314#discussion_r765726406

Comment (trantanen) [reply samples/nrf9160/modem_shell/src/main.c:146]: I added DNM label so that this won't be merged before we get the dependencies in place.

**Review by divipillai — APPROVED** [review]

File: samples/nrf9160/modem_shell/src/main.c (hunk 1 above)
Comment (lemrey) [reply samples/nrf9160/modem_shell/src/main.c:147]: @trantanen all the dependencies have been ported to the new AT APIs, I'll dismiss your review to get this merged.

**Review by rlubos — APPROVED** [review]

**Pull request merged**